### DISCUSSION TITLE
Do you have an example of cross host vm migration?

### DISCUSSION THREAD

**Discussion by nikitashamgunov** [cloud-hypervisor/cloud-hypervisor#4468] [Q&A]
@neondatabase we are considering to use cloud hypervisor to build elastic compute. for that we need to be able to live migrate VMs w/o breaking user connection to the database running inside of the VM and also preserve caches.

Do you have an example and prereq to implement such migration?

**Comment by sboeuf** [comment]
Hi @nikitashamgunov, I'm glad to hear you're interested in Cloud Hypervisor. You can find more information about live migration through our [documentation](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/live_migration.md). You might find useful taking a look at our integration tests suite since it includes live migration validation: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/tests/integration.rs#L8147-L8566
Please let us know if you run into any issue.

**Reply by likebreath** [reply]
Just want to emphasize that the ["nested-vm-migration"](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/live_migration.md#nested-vm-migration) section of our documentation is the example for cross host migration.

**Comment by liuw** [comment]
Hi Nikita. Nice to see you here.

Assuming the connection is made via network, your will need to configure your network to reroute traffic correctly. That's outside of Cloud Hypervisor's scope.

Other than that, Cloud Hypervisor replicates guest states and memory to the destination VM, so I suppose whatever cache you have in the destination VM is preserved.

There will be a short time period that the VM becomes unresponsive, because the final chunk of memory and states will be transferred with the source VM paused. I have not measured how long is this "black-out" time period. You may want to measure it yourself to make sure it meets your requirements.
